services/app-api/utils/other/other.ts

```diff
@@ -18,3 +25,80 @@ export const createReportName = (
   const fullStateName = States[state as keyof typeof States];
   return `${fullStateName} ${reportName} ${reportYear} - Period ${period}`;
 };
+
+export const lastCreatedWorkPlan = (
+  currentWorkPlans: ReportMetadataShape[]
+): ReportMetadataShape | undefined => {
+  let lastCreatedWorkPlan: ReportMetadataShape | undefined = undefined;
+
+  // For each work plan...
+  currentWorkPlans.forEach((workPlan: ReportMetadataShape) => {
+    /*
+     * ...if the workplan hasn't been used to create a SAR before AND
+     * the work plan has a status of "Not Started" OR "In Progress"...
+     */
+    if (
+      (workPlan.status === ReportStatus.NOT_STARTED ||
+        workPlan.status === ReportStatus.IN_PROGRESS) &&
+      !workPlan?.associatedSar
+    ) {
+      /*
+       * ...then do one of two things: if there are multiple work plans that meet this criteria,
+       * grab the one that was created most recently and return that as our work plan to
+       * copy from...
+       */
+      if (
+        lastCreatedWorkPlan &&
+        workPlan.createdAt > lastCreatedWorkPlan?.createdAt
+      ) {
+        lastCreatedWorkPlan = workPlan;
+      } else if (!lastCreatedWorkPlan) {
```

Comment:
suggestion: could this be simplified to 

`if (
        !lastCreatedWorkPlan ||
        workPlan.createdAt > lastCreatedWorkPlan?.createdAt
      )`